compatibility.js

```diff
@@ -39,36 +39,49 @@ function removeEolFeatures (config, version) {
  * return {object} - The converted config object
  */
 const compatFunctions = {
-    v1: (config) => {
-        // Breaking changes: minSupportedVersion key in features
+    v1: {
+        stripReasons: false,
+        compat: (config) => {
+            // Breaking changes: minSupportedVersion key in features
 
-        const v1Config = JSON.parse(JSON.stringify(config))
-        for (const feature of Object.keys(v1Config.features)) {
-            if (v1Config.features[feature].minSupportedVersion) {
-                delete v1Config.features[feature].minSupportedVersion
-                v1Config.features[feature].state = 'disabled'
+            const v1Config = JSON.parse(JSON.stringify(config))
+            for (const feature of Object.keys(v1Config.features)) {
+                if (v1Config.features[feature].minSupportedVersion) {
+                    delete v1Config.features[feature].minSupportedVersion
+                    v1Config.features[feature].state = 'disabled'
+                }
             }
-        }
 
-        return v1Config
+            return v1Config
+        }
     },
-    v2: (config) => {
-        // Breaking changes: rollout key in sub-features
+    v2: {
+        stripReasons: false,
+        compat: (config) => {
+            // Breaking changes: rollout key in sub-features
 
-        const v2Config = JSON.parse(JSON.stringify(config))
-        for (const feature of Object.keys(v2Config.features)) {
-            const subFeatures = v2Config.features[feature].features
-            if (subFeatures) {
-                for (const subFeature of Object.keys(subFeatures)) {
-                    if (subFeatures[subFeature].rollout) {
-                        delete subFeatures[subFeature].rollout
-                        v2Config.features[feature].features[subFeature].state = 'disabled'
+            const v2Config = JSON.parse(JSON.stringify(config))
+            for (const feature of Object.keys(v2Config.features)) {
+                const subFeatures = v2Config.features[feature].features
+                if (subFeatures) {
+                    for (const subFeature of Object.keys(subFeatures)) {
+                        if (subFeatures[subFeature].rollout) {
+                            delete subFeatures[subFeature].rollout
+                            v2Config.features[feature].features[subFeature].state = 'disabled'
+                        }
                     }
                 }
             }
-        }
 
-        return v2Config
+            return v2Config
+        }
+    },
+    v3: {
+        stripReasons: false,
+        compat: (config) => {
```

Comment:
It might be simpler to pass in the unmodified config as the first argument to these functions and on v3 merge back in reasons from the unmodified config.

That way if we ever have the need to do this kinda thing in future we'd not have to pass more arguments.